openmdao/vectors/vector.py

```diff
@@ -578,13 +578,19 @@ def set_var(self, name, val, idxs=_full_slice, flat=False):
             The locations where the data array should be updated.
         flat : bool
             If True, set into flattened variable.
+        var_name : str or None
+            If specified, the variable name to use when reporting errors. This is useful
+            when setting an IVC value that the user only knows by a connected input name.
         """
+        if not var_name:
```

Comment:
Since `set_var` gets called a lot, we could get rid of this block and just replace `{name}` in the error messages with `{var_name if var_name else name}`.  That way there's no additional overhead introduced unless there's an actual error.